pwndbg/commands/heap.py

```diff
@@ -182,6 +182,46 @@ def heap(addr=None, verbose=False, simple=False) -> None:
         for chunk in h:
             malloc_chunk(chunk.address, verbose=verbose, simple=simple)
 
+parser = argparse.ArgumentParser(
+    formatter_class=argparse.RawTextHelpFormatter,
+    description="""Try to find if an address is belond to a heap chunk.
+
+If yes -- print chunk. Search all heaps.""",
+)
+parser.add_argument(
+    "addr",
+    type=int,
+    help="Address of the interest.",
+)
+parser.add_argument(
+    "-v", "--verbose", action="store_true", help="Print all chunk fields, even unused ones."
+)
+parser.add_argument(
+    "-s", "--simple", action="store_true", help="Simply print malloc_chunk struct's contents."
+)
+parser.add_argument(
+    "-a", "--all", action="store_true", help="Do not stop after first hit"
+)
+
+
+@pwndbg.commands.ArgparsedCommand(parser, category=CommandCategory.HEAP)
+@pwndbg.commands.OnlyWhenRunning
+@pwndbg.commands.OnlyWithResolvedHeapSyms
+@pwndbg.commands.OnlyWhenHeapIsInitialized
+def hi(addr, verbose=False, simple=False, all=False) -> None:
+    """Iteratively search all heaps for contain current address in their chunks bodys.
+    """
+    allocator = pwndbg.heap.current
+    arenas = allocator.arenas
+
+    for arena in arenas:
+        for heap in arena.heaps:
+            if heap.start <= addr and heap.end > addr:
```

Comment:
Can you, please, clarify which one should I use?
The feature does not garant that passed addr would be start of chunk, it could be middle of chunk. So I can't get how to use it without loops